src/languageservice/parser/yamlParser07.ts

```diff
@@ -115,6 +115,12 @@ function recursivelyBuildAst(parent: ASTNode, node: Yaml.YAMLNode): ASTNode {
                 return new BooleanASTNodeImpl(parent, parseYamlBoolean(value), node.startPosition, node.endPosition - node.startPosition);
             }
 
+            //This is patch to support YAML null values: https://yaml.org/type/null.html
+            const possibleNullValues = ['~', 'NULL', 'Null', 'null'];
+            if (instance.plainScalar && possibleNullValues.indexOf(value.toString()) !== -1) {
```

Comment:
I wonder if it makes sense to just to change:
```typescript
case Yaml.ScalarType.null: {
      return new StringASTNodeImpl(parent, instance.startPosition, instance.endPosition - instance.startPosition);
}
```

to 
```typescript
case Yaml.ScalarType.null: {
     return new NullASTNodeImpl(parent, node.startPosition, node.endPosition - node.startPosition)
}
```

Reply:
I will check if the tests succeed by doing so.